From 1104cf9a3f517605f6b6dd140138a7f5d8d1b39a Mon Sep 17 00:00:00 2001
From: Suniel Mahesh <sunil@amarulasolutions.com>
Date: Wed, 5 Jul 2023 16:12:22 +0530
Subject: [PATCH] configs: imx8mp-icore: Change boot count register and enable
 WDT

Change boot count register to use SNVS GP registers, enable WDT
and use 40S as timeout.

Signed-off-by: Suniel Mahesh <sunil@amarulasolutions.com>
---
 configs/imx8mp-icore-mx8mp-ctouch2_defconfig | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/configs/imx8mp-icore-mx8mp-ctouch2_defconfig b/configs/imx8mp-icore-mx8mp-ctouch2_defconfig
index a067ed32c3..e83047b424 100644
--- a/configs/imx8mp-icore-mx8mp-ctouch2_defconfig
+++ b/configs/imx8mp-icore-mx8mp-ctouch2_defconfig
@@ -19,7 +19,7 @@ CONFIG_SPL_MMC=y
 CONFIG_SPL_SERIAL=y
 CONFIG_SPL_DRIVERS_MISC=y
 CONFIG_BOOTCOUNT_BOOTLIMIT=3
-CONFIG_SYS_BOOTCOUNT_ADDR=0x30390078
+CONFIG_SYS_BOOTCOUNT_ADDR=0x30370090
 CONFIG_SPL_STACK=0x960000
 CONFIG_SPL=y
 CONFIG_SYS_BOOTCOUNT_SINGLEWORD=y
@@ -33,7 +33,6 @@ CONFIG_SPL_LOAD_FIT=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_DEFAULT_FDT_FILE="imx8mp-icore-mx8mp-ctouch2-of10.dtb"
-CONFIG_CYCLIC=y
 CONFIG_BOARD_LATE_INIT=y
 CONFIG_SPL_MAX_SIZE=0x26000
 CONFIG_SPL_HAS_BSS_LINKER_SECTION=y
@@ -64,6 +63,7 @@ CONFIG_CMD_FUSE=y
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
+CONFIG_CMD_WDT=y
 CONFIG_CMD_CACHE=y
 CONFIG_CMD_REGULATOR=y
 CONFIG_CMD_EXT4_WRITE=y
@@ -115,4 +115,8 @@ CONFIG_MXC_UART=y
 CONFIG_SYSRESET=y
 CONFIG_SPL_SYSRESET=y
 CONFIG_SYSRESET_PSCI=y
-# CONFIG_SPL_WDT is not set
+CONFIG_SYSRESET_WATCHDOG=y
+CONFIG_SYSRESET_WATCHDOG_AUTO=y
+# CONFIG_WATCHDOG is not set
+CONFIG_WATCHDOG_TIMEOUT_MSECS=40000
+CONFIG_IMX_WATCHDOG=y
-- 
2.25.1

